MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -156,35 +158,51 @@ -(void) saveSpotifyData:(NSDictionary *) spotifyData{
     PFUser *current = [PFUser currentUser];
     PFQuery *query = [PFQuery queryWithClassName:@"Music"];
     [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
-        if (![[results valueForKey:@"userId"]containsObject:current.objectId]){
-            //Checks if User has already been added to database before adding
-            music[@"userId"] = current.objectId;
+        [User user].artists = [NSArray arrayWithArray:spotifyData[@"images"]];
+        if (![[[results valueForKey:@"author"] valueForKey:@"objectId"] containsObject: current.objectId]){
+            music[@"author"] = current;
             music[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
             music[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
             music[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
             music[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
             music[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
-            music[@"userimage"] = self.profilePicture;
-            music[@"username"] = self.name;
+            [self getSpotifyData:@"https://api.spotify.com/v1/me" completion:^(NSDictionary * userDict, NSError * error) {
```

Comment:
Now stores the data that was in previously SaveSpotifyData to this function to store the profile name and picture to the database for easier and simplistic way of storing data

Reply:
I notice that there is still a `[self getSpotifyData:@"https://api.spotify.com/v1/me"` call in saveSpotifyData though. Did you mean to remove that one?